app/controllers/app/base_controller.rb

```diff
@@ -3,4 +3,11 @@
 
 class App::BaseController < ApplicationController
     before_action :authenticate_user!
+
+    def check_current_user
+      if current_user == nil  
+        flash[:alert] = "You have to be logged in first"
+        redirect_to new_user_session_url
+      end 
+    end
```

Comment:
Here you could use a "guard" or "guard clause"?
Sharing a short reading in case you never heard about it: https://en.wikipedia.org/wiki/Guard_(computer_science)

```suggestion
    def check_current_user
      return unless current_user
      
      flash[:alert] = "You have to be logged in first"
      redirect_to new_user_session_url
    end
```